src/main/java/br/com/sbk/sbking/core/Deal.java

```diff
@@ -37,6 +38,8 @@ public Deal(Board board, Ruleset ruleset, Direction leader) {
         this.board = board;
         this.ruleset = ruleset;
         this.currentPlayer = leader;
+        this.board.setCardComparator(ruleset.getComparator());
```

Comment:
You setted the comparator in line 41, so I don't think it should be necessary to pass the same comparator in line 42 as a parameter

Reply:
I think I understood why are you accepting the comparator as a parameter to sort the hands. It is because the sorting will change after each deal. 
But thinking in OOP terms, I don't think this is a task that Board should handle. In a real game, the board comes and then the ruleset is setted, so the players organize the cards. But I don't think a button to sort the cards would be interesting here, so I think the deal should execute this task.

Reply:
Done that. `.sortAllHands()` has no args now and you would have to set the comparator first, if you don't want the `Board` to decide its own comparator.

About the real game, a sort action is **initiated** by a Player but **happens** in its Hand. That is why the method that really does the sorting is `Hand::sort(Comparator<Card>)`. The others are just intermediate steps.

To make the responsibilities right, I just completely removed the field `comparator` from the `Board`. A `Board` has no default ordering by itself. Like the `Hand`, it need to receive that as a parameter to sort.
Because of that, the responsibility of choosing a `Comparator<Card>` went up the line, to the places where a `Ruleset` is already defined or has more information on the wanted sorting, like the `Deal` or the `CagandoNoBequinhoGame`